.github/workflows/test.yaml

```diff
@@ -11,39 +11,18 @@ on:
 
 jobs:
   run-tests:
-    runs-on: ubuntu-20.04
+    runs-on: self-hosted
     steps:
-      - name: Install dependencies
-        run: |
-          sudo apt-get update
-          sudo apt install -y xdelta3
-
-      - name: Setup go version
-        uses: actions/setup-go@v2
-        with:
-          # Using the lastest available go version for focal
-          go-version: 1.13.8
+      - uses: actions/checkout@v2
 
-      - name: Set environment
+      - name: Run tests
         run: |
-          echo "GOHOME=${{ github.workspace }}" >> "$GITHUB_ENV"
-          echo "GOPATH=${{ github.workspace }}" >> "$GITHUB_ENV"
+          spread google:
 
-      - name: Checkout project
-        uses: actions/checkout@v2
-        with:
-          path: src/github.com/snapcore/spread
-
-      - name: Build spread
-        run: |
-          cd ${{ github.workspace }}/src/github.com/snapcore/spread
-          go get ./cmd/spread
-          
-      - name: run tests
-        env:
-          SPREAD_GOOGLE_KEY: ${{ secrets.SPREAD_GOOGLE_KEY }}
+      - name: Discard spread workers
+        if: always()
```

Comment:
I assume we need this to ensure that even for failed spread runs the cleanup runs?

Reply:
@mvo5 thanks for the review
Yes, we use this to make sure the machines are discarded in case spread cannot discard them.